### DISCUSSION TITLE
Dealing with IPv6 + disabled Port Forwarding

### DISCUSSION THREAD

**Discussion by bradut** [GerryDazoo/Slinger#220] [Ideas]
I have tried to enable port forwarding in the networks of my two remote SlingBoxes in Europe.
Unfortunately, both providers, Orange and Vodafone, only offer IPv6, and **disabled port forwarding** in their routers.

I wonder if it is possible to overcome these limitations and access those SlingBoxes from Canada after 9 Nov.2022?

I believe that SlingBox is continuously observing the server www.slingbox.com for incoming requests, and reacts by streaming to whatever player [desktop. web, mobile] is indicated via the server.

If this is the case, would it be possible to implement a similar approach to Slinger?
I mean: to run Slinger on devices located on those networks, and to indicate Slinger to listen to another server.

(I've searched for an answer into the other posts which mention IPv6, but couldn't find anything specific.)

**Comment by blasco403** [comment]
I have a similar issue, I get around it by connecting my remote network with the local network the slingbox's use via a network to network VPN with Wireguard. If one side of your connection allows port forwarding, your non port forward capable network would be setup to connect outbound to the other.

I use a RP4 at both ends.

**Comment by GerryDazoo** [comment]
Well, that's not quite how it works but what you're asking for is way out of scope for what I'm willing to support anytime soon. blasco403 has a reasonable solution. Just wondering if you can supply your own non-brain damaged router to use instead of the ISP supplied?

**Reply by bradut** [reply]
Thank you Gerry,
I will try the VPN solution indicated by blasco403
I don't know if I can use a different, non-brain damaged router, because those modem-routers are provided and controlled by ISPs.

About the Vodafone router, I've read in forums that they have updated the firmware and switched to IPv6 this summer, w/o even letting their users know.
About Orange, it seems that it may be possible to go back to IPv4 + Port Forwarding by paying a monthly fee of about 2 Euros.

**Reply by GerryDazoo** [reply]
Firstly: Bastards!

This is a little off-topic but I recently worked with someone who also had an issue with not being able to receive any traffic from the Internet. Good news for him is he had access to the low-level command line of his routers. He could ssh into the router and get a command line prompt.  I think they were ASUS but many router support this.
So what we did was setup a script on the remote to SSH into his home router that setup a reverse tunnel to allow him to connect back into the problemic ISP. So instead of suppling the IP address of the remote system he'd connect to his home router on the SSH tunnel port.

Not sure this applicable to your European issues but many of the ISPs will give you a "Modem" integrated with a "Router" and charge you a monthly fee for the pleasure. In many cases you can supply your on devices. I've got Internet from a cable provider (Comcast/Xfinity) and they provisioned my cable modem and I use my own router. Similarly, I did the same with a fiber connection from ATT. The supplied the modem and gave me the bits from the Internet via Ethernet which I plugged into the uplink of my own router.

**Reply by tom76017** [reply]
I just spent 4 weeks in Corpus Christi with an OpenVPN connection to my ASUS router in Fort Worth. Samba and Slinger never failed.  Check out https://www.asuswrt-merlin.net/

**Reply by bradut** [reply]
Just wanted to give an update: I asked both cable companies to enable dynamic IPv4 and this solved my problem.
